plugins/pixart-rf/fu-pxi-device.c

```diff
@@ -126,7 +149,32 @@ fu_pxi_device_calculate_checksum (const guint8 *data, gsize sz)
 	return checksum;
 }
 
-/* FIXME: is @port required? */
+static gboolean
+fu_pxi_device_check_support_resume (FuPxiDevice *self,
+				 	FuFirmware *firmware,
+				  	GError **error)
+{
+	g_autoptr(GBytes) fw = NULL;
+	g_autoptr(GPtrArray) chunks = NULL;
+	guint16 checksum_tmp = 0x0;
+	
+	/* get the default image */
+	fw = fu_firmware_get_image_default_bytes (firmware, error);
+	chunks = fu_chunk_array_new_from_bytes (fw, 0x0, 0x0, FU_PXI_DEVICE_OBJECT_SIZE_MAX);
+	
+	/* calculate device current checksum */
+	for (guint i = 0; i < self->offset; i++) {	
+		FuChunk *chk = g_ptr_array_index (chunks, i);
+		checksum_tmp += fu_pxi_device_calculate_checksum (chk->data, chk->data_sz);
+	}
+	
+	/* check current file is different with previous fw bin or not */
+	if (self->checksum != checksum_tmp)
+	        return FALSE;
```

Comment:
if you return FALSE you need to set the GError

Reply:
Ok I will fix right now